packages/wire/__tests__/gossip/DnsPeerQuery.spec.ts

```diff
@@ -0,0 +1,139 @@
+                r: 0,
+                a: 6,
+                n: 25,
+                dnsSeed: "lseed.bitcoinstats.com",
+            });
+
+            expect(resolver.resolveSrv.getCall(0).args[0]).to.equal(
+                `r0.a6.n25.lseed.bitcoinstats.com`,
+            );
+        });
+
+        it("should return peers using information from resolver", async () => {
+            var resolver = {
+                resolveSrv: Sinon.stub().returns([
+                    {
+                        name:
+                            "ln1qwktpe6jxltmpphyl578eax6fcjc2m807qalr76a5gfmx7k9qqfjwy4mctz.lseed.bitcoinstats.com.",
+                        port: 6331,
+                    },
+                ]),
+                resolve: Sinon.stub().returns(["139.59.143.87"]),
+            };
+
+            var dnsPeerQuery = new DnsPeerQuery(logger, resolver);
+
+            await expect(
+                dnsPeerQuery.query({
+                    dnsSeed: "lseed.bitcoinstats.com",
+                }),
+            ).to.eventually.have.same.deep.members([
+                {
+                    address: "139.59.143.87",
+                    port: 6331,
+                    publicKey: bech32Decode(
+                        "ln1qwktpe6jxltmpphyl578eax6fcjc2m807qalr76a5gfmx7k9qqfjwy4mctz",
+                    ),
+                },
+            ]);
+
+            expect(resolver.resolveSrv.getCall(0).args[0]).to.equal("lseed.bitcoinstats.com");
+            expect(resolver.resolve.getCall(0).args[0]).to.equal(
+                "ln1qwktpe6jxltmpphyl578eax6fcjc2m807qalr76a5gfmx7k9qqfjwy4mctz.lseed.bitcoinstats.com.",
+            );
+        });
+
+        it("should filter out failed ip resolutions", async () => {
+            var resolveStub = Sinon.stub();
+
+            resolveStub
+                .onFirstCall()
+                .returns(Promise.resolve(["139.59.143.87"]))
+                .onSecondCall()
+                .returns(Promise.reject(new Error()));
+
+            var resolver = {
+                resolveSrv: Sinon.stub().returns(
+                    Promise.resolve([
+                        {
+                            name:
+                                "ln1qwktpe6jxltmpphyl578eax6fcjc2m807qalr76a5gfmx7k9qqfjwy4mctz.lseed.bitcoinstats.com.",
+                            port: 6331,
+                        },
+                        {
+                            name:
+                                "ln1qv2w3tledmzczw227nnkqrrltvmydl8gu4w4d70g9td7avke6nmz2tdefqp.lseed.bitcoinstats.com.",
+                            port: 9735,
+                        },
+                    ]),
+                ),
+                resolve: resolveStub,
+            };
+
+            var dnsPeerQuery = new DnsPeerQuery(logger, resolver);
+
+            await expect(
+                dnsPeerQuery.query({
+                    dnsSeed: "lseed.bitcoinstats.com",
+                }),
+            ).to.eventually.have.same.deep.members([
```

Comment:
Would rather assert using `deep.equals`.   We want to capture that the failed error did not result in a record.  Similarly, a we should test a bad bech32 decoding and make sure it doesn't appear in results.